### ISSUE TITLE
MLC Chat Failure with TVM Error: Incorrect Number of Arguments (Android)

### ISSUE THREAD

**Issue by Msatisfied-agi** [mlc-ai/mlc-llm#2753]
## 🐛 Bug
I encountered  MLC Chat failed error while trying to use the MLC Chat app in android. The app fails to following AI models:
1-Gemma 2b
2-Redpajama incite 3b
3-Mistral 7b
4-Llama3 8b 
(Only Llama-2-7b and phi-2 work)

### Tested Devices:

Device 1: OnePlus Ace 2 Pro

Processor: Snapdragon 8 Gen 2
RAM: 8 GB

Device 2: Infinix Note 30

Processor: MediaTek Helio G99
RAM: 8 GB

### Steps to Reproduce:

Download and install the MLCChat app on either of the above devices.
Attempt to use any AI model except Phi2 and Llama2 by first downloading it and then initializing it while initialization this happens.
All of stated ai models except phi2 and Llama2 consistently fails with the same error message on both devices.
I have tried deleting and downloading the models multiple times, but the issue persists.

**Expected Behavior**:

These Ai models should initialize and start ready to chat.

### Actual Behavior:

Gives this error of MLC Chat failed.

**Stack trace**:

org.apache.tvm.Base$TVMError: TVMError: Function vm.builtin.paged_attention_kv_cache_create_reduced(0: runtime.ShapeTuple, 1: int64_t, 2: int64_t, 3: int64_t, 4: int64_t, 5: int, 6: double, 7: double, 8: runtime.NDArray, 9: runtime.PackedFunc, 10: runtime.PackedFunc, 11: runtime.PackedFunc, 12: runtime.PackedFunc, 13: runtime.PackedFunc, 14: runtime.PackedFunc, 15: runtime.PackedFunc, 16: runtime.PackedFunc, 17: runtime.PackedFunc, 18: runtime.PackedFunc) -> relax.vm.AttentionKVCache expects 19 arguments, but 18 were provided.
Stack trace:
  File "/Users/kartik/mlc/tvm/include/tvm/runtime/packed_func.h", line 1908

 at org.apache.tvm.Base.checkCall(Base.java:173)
 at org.apache.tvm.Function.invoke(Function.java:130)
 at ai.mlc.mlcllm.ChatModule.reload(ChatModule.java:46)
 at ai.mlc.mlcchat.AppViewModel$ChatState$mainReloadChat$1$2.invoke(AppViewModel.kt:648)
 at ai.mlc.mlcchat.AppViewModel$ChatState$mainReloadChat$1$2.invoke(AppViewModel.kt:646)
 at ai.mlc.mlcchat.AppViewModel$ChatState.callBackend(AppViewModel.kt:548)
 at ai.mlc.mlcchat.AppViewModel$ChatState.mainReloadChat$lambda$3(AppViewModel.kt:646)
 at ai.mlc.mlcchat.AppViewModel$ChatState.$r8$lambda$CXL6v4mjTu_Sr5Pk2zFDcus0R-8(Unknown Source:0)
 at ai.mlc.mlcchat.AppViewModel$ChatState$$ExternalSyntheticLambda2.run(Unknown Source:8)
 at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
 at java.util.concurrent.FutureTask.run(FutureTask.java:264)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
 at java.lang.Thread.run(Thread.java:1012)

**Error message**:

TVMError: Function vm.builtin.paged_attention_kv_cache_create_reduced(0: runtime.ShapeTuple, 1: int64_t, 2: int64_t, 3: int64_t, 4: int64_t, 5: int, 6: double, 7: double, 8: runtime.NDArray, 9: runtime.PackedFunc, 10: runtime.PackedFunc, 11: runtime.PackedFunc, 12: runtime.PackedFunc, 13: runtime.PackedFunc, 14: runtime.PackedFunc, 15: runtime.PackedFunc, 16: runtime.PackedFunc, 17: runtime.PackedFunc, 18: runtime.PackedFunc) -> relax.vm.AttentionKVCache expects 19 arguments, but 18 were provided.
Stack trace:
  File "/Users/kartik/mlc/tvm/include/tvm/runtime/packed_func.h", line 1908

**Comment by MasterJH5574** [comment]
@mengshyu Could you help take a look?

**Comment by mengshyu** [comment]
Hi @Msatisfied-agi, I verified all default models on the Samsung S23, and all models work well. I wonder when you installed the Android APK. We had an update on 7/31/2024. Can you try it again if you are not using the latest version? Thanks.

Demo App: https://llm.mlc.ai/docs/deploy/android.html

**Comment by Msatisfied-agi** [comment]
Can you confirm the app version number or something? I have downloaded the app again from website and reinstalled it but it still shows version number 1.0 in setting as it was showing before. Anyway I downloaded it again and now downloading ai model again. I will update on situation later.

**Comment by Msatisfied-agi** [comment]
So, yeah. App was downloaded a month and more before. I updated it now and its working fine. No issue. I did sent this error message to chatgpt a month before, that also said about app version conflict but at that time, I downloaded it again, it was same version as before at that time, so faced this issue for long time. Heads up to you guys who updated demo app a week before and I was not are of it. Now, it works well. Thanks.
